src/server/page_server.cpp

```diff
@@ -110,10 +110,21 @@ void page_server::receive_data_page_fetch (cubpacking::unpacker &upk)
   upk.unpack_string (message);
 
   VPID vpid;
-  std::memcpy (&vpid, message.c_str (), sizeof (vpid));
+  int bytes_read = 0;
+  std::memcpy (&vpid.pageid, message.c_str () + bytes_read, sizeof (vpid.pageid));
+  bytes_read += sizeof (vpid.pageid);
+
+  std::memcpy (&vpid.volid, message.c_str () + bytes_read, sizeof (vpid.volid));
+  bytes_read += sizeof (vpid.volid);
 
   LOG_LSA target_repl_lsa;
-  std::memcpy (&target_repl_lsa, message.c_str () + sizeof (vpid), sizeof (target_repl_lsa));
+  std::memcpy (&target_repl_lsa, message.c_str () + bytes_read, sizeof (target_repl_lsa));
+  bytes_read += sizeof (target_repl_lsa);
+
+  _er_log_debug (ARG_FILE_LINE,
```

Comment:
The debug lines should be removed. They were used just for debugging. And `bytes_read` goes down too.